zerver/lib/import_realm.py

```diff
@@ -1995,3 +2008,54 @@ def add_users_to_system_user_groups(
         )
         for membership in usergroup_memberships
     )
+
+
+def check_migration_status(migration_status_file: str) -> None:
+    # Each server might have a different set of apps installed (e.g., Zulip Cloud has
+    # zilencer/corporate, but self-hosted might not). This function only checks for
+    # migration mismatches in apps installed on both servers.
+    local_migrations = parse_migration_status(get_migration_status())
+    exported_migrations = parse_migration_status(migration_status_file)
+
+    mismatched_migrations_log = [
+        f"Migrations for the '{app}' app doesn't match; showing diff.\n"
+        + diff_strings(local_migrations[app], exported_migrations[app])
+        for app in exported_migrations
+        if local_migrations.get(app) and local_migrations[app] != exported_migrations[app]
+    ]
+
+    match = re.search(r"ZULIP_VERSION=([^\s]+)", migration_status_file)
+    exported_zulip_version = match.group(1) if match else "Unknown"
+
+    if mismatched_migrations_log:
+        raise AssertionError(
+            f"""
+---
+Mismatch detected in applied migrations between the exported file and this server.
+Ensure both servers are running the same Zulip version!
+
+Zulip versions:
+* Exported realm: {exported_zulip_version}
+* This server: {ZULIP_VERSION}
+---
+"""
+            + "\n".join(mismatched_migrations_log)
+        )
+
+
+def parse_migration_status(text: str) -> dict[str, str]:
```

Comment:
This should be the function before the one that uses it; code is more readable that way.